src/main/java/ch/njol/skript/expressions/ExprReplace.java

```diff
@@ -0,0 +1,133 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptConfig;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.ReplacementTypes;
+import ch.njol.util.Kleenean;
+
+
+@Name("Replace")
+@Description("Replaces all occurrences of a given text with another text and returns the replaced text.")
+@Examples({"on chat:",
+		"\tset {_hey} to replace all \"hello\" in message with \"hey\" #this will replace all the values without changing the message",
+		"\tsend {_hey}",
+		"set {_no} to  replace first \"yes\" in \"Yes, yes\" with \"no\" with case sensitivity #Only the second yes gets replaced with no"
+})
+@Since("INSERT VERSION")
+public class ExprReplace extends SimpleExpression<String> {
+	static {
+		Skript.registerExpression(ExprReplace.class, String.class, ExpressionType.COMBINED,
+				"replace (1¦first|2¦last|0¦all|every|) %strings% with %string% in %string% [(4¦with case sensitivity)]",
+				"regex replace (1¦first|2¦last|0¦all|every|) %strings% with %string% in %string%");
+	}
+	
+	private ReplacementTypes type = ReplacementTypes.ALL;
+	@SuppressWarnings("null")
+	private Expression<String> textToReplace, replacement, storage;
+	private boolean caseSensitive = false;
+	private boolean regex;
+	
+	@Override
+	@SuppressWarnings({"unchecked", "null"})
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean kleenean, SkriptParser.ParseResult parseResult) {
+		int mark = parseResult.mark;
+		if (matchedPattern == 1) regex = true;
+		if (SkriptConfig.caseSensitive.value() || (parseResult.mark ^ 4) < 4)
+			caseSensitive = true;
+		textToReplace = (Expression<String>) expressions[0];
+		replacement = (Expression<String>) expressions[1];
+		storage = (Expression<String>) expressions[2];
+		type = ReplacementTypes.values()[mark < 4 ? mark : (mark ^ 4)];
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	@SuppressWarnings("null")
+	protected String[] get(Event event) {
+		Object[] oldtextToReplace = Arrays.stream(textToReplace.getAll(event)).map((String he) -> ((caseSensitive && !regex ? "" : "(?i)") + (regex ? he : Pattern.quote(he)))).toArray();
```

Comment:
I think you can just do this and then later on in your switch statement loop over strings instead of having to cast
```suggestion
		String[] oldtextToReplace = Arrays.stream(textToReplace.getAll(event)).map((String he) -> ((caseSensitive && !regex ? "" : "(?i)") + (regex ? he : Pattern.quote(he)))).toArray(String[]::new);
```